src/pages/notifications/index.vue

```diff
@@ -301,26 +305,19 @@ export default {
         // No-op
       }
     }
-    window.addEventListener('beforeunload', this.updateEventLastSeenTs);
+    this.$api.$post(updateEventLastSeen());
+    this.lastEntryTs = Date.now();
   },
   // For SPA navigation
   beforeRouteLeave(to, from, next) {
-    this.updateEventLastSeenTs();
-    window.removeEventListener('beforeunload', this.updateEventLastSeenTs);
+    this.updateEventLastSeenTs(this.lastEntryTs || Date.now());
```

Comment:
Just call `this.updateEventLastSeenTs(this.lastEntryTs)` on `mounted` or maybe in `handleRefresh `?

Reply:
Also we should call `this.$api.$post(updateEventLastSeen());` in `handleRefresh`
maybe refactor 
```
this.updateEventLastSeenTs(Date.now());
this.$api.$post(updateEventLastSeen());
```
into a method